Automattic/simplenote-ios
# Notes List: Swifting UITableViewDatasource

### Details:
In this PR we're reimplementing UITableViewDatasource conformance in Swift
The Swift realm is gonna make things muuuuch more elegant when initializing multiple entity kinds (and hopefully things will be more compact).

Ref. #507 
cc @bummytime (Thanks in advance!!)

### Test
1. Log into your account
2. Verify your notes list loads properly
3. Verify the filters (Tags / Trash) work as expected
4. Verify that swiping over any row displays actions onscreen

### Release
These changes do not require release notes.
